tests/infra/network.py

```diff
@@ -173,12 +181,23 @@ def _add_node(self, node, lib_name, args, target_node=None, recovery=False):
                 timeout=args.ledger_recovery_timeout if recovery else 3
             )
 
+        snapshot_dir = None
+        if from_snapshot:
+            LOG.info("Joining from snapshot")
+            snapshot_dir = target_node.get_snapshots()
+            # For now, we must have a snapshot to resume from when attempting
```

Comment:
That's for now and for the foreseeable future.

Reply:
I think that eventually, joining from a snapshot will become the default when joining a new node in our Python infra. If the `target_node` has no snapshot, then we'll join from append entries since the beginning of time.